src/control/BarrettHandKinematicSimulationPositionCommandExecutor.cpp

```diff
@@ -121,42 +133,45 @@ std::future<void>
 BarrettHandKinematicSimulationPositionCommandExecutor::execute(
     const Eigen::VectorXd& goalPositions)
 {
-  std::lock_guard<std::mutex> lockSpin(mMutex);
+  std::lock_guard<std::mutex> lock(mMutex);
 
-  if (mInExecution)
-    throw std::runtime_error("Another command in execution.");
+  if (mInProgress)
+    throw std::runtime_error("Another position command is in progress.");
 
   if (goalPositions.size() != 4)
   {
     std::stringstream message;
-    message << "GoalPositions must have 4 elements, but ["
+    message << "BarrettHand goal positions must have 4 elements, but ["
             << goalPositions.size() << "] given.";
     throw std::invalid_argument(message.str());
   }
 
   mPromise.reset(new std::promise<void>());
-  mProximalGoalPositions = goalPositions.head<3>();
-  mSpreadGoalPosition = goalPositions.row(3);
-  mInExecution = true;
+
+  Eigen::VectorXd proximalGoalPositions = goalPositions.head<3>();
```

Comment:
Nit: It would be good to use fixed-size vector when we know the size in compile time. So `Eigen::Vector3d proximalGoalPositions = goalPositions.head<3>();`